mailet/api/src/main/java/org/apache/mailet/Mail.java

```diff
@@ -98,7 +99,9 @@
      * @throws MessagingException when an error occurs while retrieving the message
      */
     MimeMessage getMessage() throws MessagingException;
-    
+
+    Message getParsedMimeMessage() throws MessagingException;
```

Comment:
I do not see in what "or it can need to be reparsed" is related with `Optional`.
`it can be not parsed` I don't think this will happen often also in this case I think we can be happy by throwing a MessagingError